pkg/image/registry/etcd/etcd.go

```diff
@@ -82,10 +84,39 @@ func (r *Etcd) CreateImage(ctx kapi.Context, image *api.Image) error {
 		return err
 	}
 
-	err = r.CreateObj(key, image, 0)
+	err = r.AtomicUpdate(key, &api.Image{}, func(obj runtime.Object) (runtime.Object, error) {
+		existing, ok := obj.(*api.Image)
+		if !ok {
```

Comment:
The docs for AtomicUpdate say this type check is guaranteed to return the type specified as the 2nd arg. I'm certainly fine keeping this ok check here, but I think you removed similar checks in the rest storage methods below, so I thought I'd bring it up.